src/AI/Muriel.java

```diff
@@ -139,7 +139,7 @@ private UnitMatchupAndMetaInfo getUnitMatchupInfo(Unit myUnit, Unit otherUnit)
     if( null != umami ) return umami;
 
     double myDamage = 0;
-    WeaponModel myWeapon = myUnit.chooseWeapon(otherUnit.model, 1, false);
+    WeaponModel myWeapon = myUnit.chooseWeapon(otherUnit.model, 1, myUnit.model.hasMobileWeapon());
```

Comment:
This is a fundamental change to how Muriel calculates unit matchups.

Reply:
Was this to accommodate antitanks? Won't Muriel just ignore them either way?
I think this might only make a difference for BShips, actually.

Reply:
It is to accommodate antitanks. I made the smallest possible change to avoid crashing from null ref exception.

This is why, years ago, I asked if `hasMobileWeapon` was what you really wanted when you wrote `hasDirectWeapon`. You said no.

Reply:
I meant it: `hasDirectWeapon` is what I intended. I deliberately kept Muriel from knowing about ranged attacks until I could decide how exactly to handle them.

Reply:
I'm well aware, but then you did not guard against *buying* antitanks... and in any case, you could start with them on the map and you'd still crash without this change.

Reply:
Well, maybe I need to review that as part of whatever upgrades I ever find time for one day.

We can leave it like this for now; I don't think the change will affect much, practically speaking.